### PR TITLE
Minor enhancement on RedisRegistrationStore

### PR DESCRIPTION
**Pull request by sbernard31**
Fix some potential race condition issue an add kind of clemency for expiration.

### CODE REVIEW

File: leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java [hunk 1]
```diff
@@ -62,6 +62,11 @@
  */
 public class RedisRegistrationStore implements CaliforniumRegistrationStore, Startable, Stoppable {
 
+    /** Default time in seconds between 2 cleaning tasks (used to remove expired registration). */
+    public static final long DEFAULT_CLEAN_PERIOD = 60;
+    /** Defaut Extra time for registration lifetime in seconds */
+    public static final long DEFAULT_CLEMENCY = 0;
```
Comment (msangoi) [inline leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java]: what do you think of renaming "clemency" to "grace period"? again a naming question ;)

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/Registration.java [hunk 2]
```diff
@@ -262,7 +262,11 @@ public Date getLastUpdate() {
     }
 
     public boolean isAlive() {
-        return lastUpdate.getTime() + lifeTimeInSec * 1000 > System.currentTimeMillis();
+        return isAlive(0);
+    }
+
+    public boolean isAlive(long clemencyInSec) {
```
Comment (msangoi) [inline leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/Registration.java]: I think this method needs a proper javadoc, in order to understand the meaning of this grace/clemency period.
And the default `isAlive()` as well.

File: leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java (hunk 1 above)
Comment (sbernard31) [reply leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java]: I will change it

File: leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/Registration.java (hunk 2 above)
Comment (sbernard31) [reply leshan-server-core/src/main/java/org/eclipse/leshan/server/registration/Registration.java]: I can try

**Pull request merged**